junebug/api.py

```diff
@@ -215,10 +215,17 @@ def send_message(self, request, body, channel_id):
             raise ApiUsageError(
                 'Only one of "from" and "reply_to" may be specified')
 
-        try:
-            self.service.getServiceNamed(channel_id)
-        except KeyError:
-            raise ChannelNotFound()
+        channel = yield Channel.from_id(
+            self.redis, self.config, channel_id, self.service)
+
+        if (channel.character_limit is not None
+                and len(body['content']) > channel.character_limit):
+            raise ApiUsageError(
+                'Message content %r is of length %d, which is greater than the'
+                ' character limit of %d' % (
+                    body['content'], len(body['content']),
+                    channel.character_limit)
+                )
```

Comment:
I'm wondering if this logic should live in the `Channel.send_*` messages. At the moment `Channel` seems to contain the logic for how to use its config properties.